src/normal.c

```diff
@@ -4917,6 +4914,8 @@ nv_replace(cmdarg_T *cap)
 	}
 	else
 	{
+	    char_u *ptr = ml_get_cursor();
```

Comment:
is it required to get the ptr here ?

Reply:
No it's not. I brought it forward from the previous code.
Having said that, we need to make sure that ptr after the for loop because it is used unconditionally after the loop if netbeans_active() returns TRUE.

Here is a modification that might be better:
`	    char_u *ptr;

	    // Replace the characters within one line.
	    for (n = cap->count1; n > 0; --n)
	    {
		// Get ptr again, because ins_copychar() and showmatch()
		// will have released the line.
		// At the same time we let know that the line will be changed.
		if (cap->nchar == Ctrl_E || cap->nchar == Ctrl_Y)
		{
		  int c = ins_copychar(curwin->w_cursor.lnum
					   + (cap->nchar == Ctrl_Y ? -1 : 1));

		  ptr = ml_get_buf(curbuf, curwin->w_cursor.lnum, TRUE);
		  if (c != NUL)
		    ptr[curwin->w_cursor.col] = c;
		}
		else
		{
		    ptr = ml_get_buf(curbuf, curwin->w_cursor.lnum, TRUE);
		    ptr[curwin->w_cursor.col] = cap->nchar;
		}
		if (p_sm && msg_silent == 0)
		    showmatch(cap->nchar);
		++curwin->w_cursor.col;
	    }
#ifdef FEAT_NETBEANS_INTG
	    if (netbeans_active())
	    {
		colnr_T  start = (colnr_T)(curwin->w_cursor.col - cap->count1);
		ptr = ml_get_buf(cufbuf, curwin->w_cursor.lnum);

		netbeans_removed(curbuf, curwin->w_cursor.lnum, start,
							   cap->count1);
		netbeans_inserted(curbuf, curwin->w_cursor.lnum, start,
					       &ptr[start], (int)cap->count1);
	    }
#endif
`

Reply:
thanks, I think this is better and closer to the current version anyhow.  However, unless I am missing something, you don't need the single declaration `char_u *ptr;`  at the start of the `else` condition. It would be shadowing the existing `ptr` variable.

Reply:
ah, now I see it, you removed that `ptr` declaration at the beginning of the function, in which case this is fine. Thanks.